aws/amazon-ecs-cli
# fix ecs-cli up --tags parser, had bugs with allowed symbols

correct tag parsing with `=` to bring ecs-cli into compliance with AWS tag standards
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html#tag-restrictions

```bash
ecs-cli up \
  --tags 'foo-bar=orange=false red-apple=standard,lemon2=yellow'
```
should result in two EC2 tags of
```
tag         value
---------------------
foo-bar     orange=false red-apple=standard
lemon2      yellow
```
Partial fix for larger issue https://github.com/aws/amazon-ecs-cli/issues/952
This PR also includes a new test case to test the `=` and other symbols scenario.

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

### Testing
- [] Integration tests passed _(unknown...will watch below in github for travis)_
- [X] Unit tests added for new functionality
- [X] Listed manual checks and their outputs in the comments
- [X] Link to issue or PR

----
### Manual Testing

I created a new cluster and instance with the following

```sh
ecs-cli up \
  --cluster-config frontend \
  --tags 'ec2-launch-mutate=monitor=false cpu-credits=standard,testtag2=lemon'
```

The changed code successfully parsed the two tag key/value pairs and corrected saved on the EC2 instance as seen in the console UI

![image](https://user-images.githubusercontent.com/679350/70578375-d086aa80-1bad-11ea-8812-cf3f636e233c.png)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
